File: packages/mm_editor/android/src/main/com/minutemastery/mm_editor/bridge/MethodHandlers.kt
# packages/mm_editor/android/src/main/com/minutemastery/mm_editor/bridge/MethodHandlers.kt

package com.minutemastery.mm_editor

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import org.json.JSONObject

class MethodHandlers {

    fun handle(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {

            "probeMedia" -> {
                val uri = call.argument<String>("uri")
                if (uri == null) {
                    result.success(error("INVALID_ARGUMENT", "Missing uri"))
                    return
                }

                val media = MediaProbe.probe(context, uri)
                result.success(success(mapOf("media" to media)))
            }

            "exportProject" -> {
                val project = call.argument<String>("project")
                val export = call.argument<Map<String, Any>>("export")

                if (project == null) {
                    result.success(error("INVALID_ARGUMENT", "Missing project"))
                    return
                }

                val output = ExportEngine.export(
                    context,
                    project,
                    context.cacheDir.absolutePath
                )

                result.success(success(mapOf("output" to output)))
            }

            else -> result.notImplemented()
        }
    }

    private fun success(payload: Map<String, Any?>): Map<String, Any?> {
        return mapOf("ok" to true) + payload
    }

    private fun error(code: String, message: String): Map<String, Any?> {
        return mapOf(
            "ok" to false,
            "error" to mapOf(
                "code" to code,
                "message" to message
            )
        )
    }
}

File: packages/mm_editor/ios/Classes/bridge/MethodHandlers.swift
# packages/mm_editor/ios/Classes/bridge/MethodHandlers.swift

import Flutter
import UIKit

class MethodHandlers {

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {

        switch call.method {

        case "probeMedia":
            guard let args = call.arguments as? [String: Any],
                let uri = args["uri"] as? String else {
                result(error(code: "INVALID_ARGUMENT", message: "Missing uri"))
                return
            }

            let media = MediaProbe.probe(uriString: uri)
            result(success(["media": media]))

        case "exportProject":
            // TODO: Implement AVAssetExportSession pipeline
            result(success([
                "output": [
                    "uri": "file:///tmp/mm_export.mp4",
                    "duration_ms": 0
                ]
            ]))

        default:
            result(FlutterMethodNotImplemented)
        }
    }

    private func success(_ payload: [String: Any]) -> [String: Any] {
        var base: [String: Any] = ["ok": true]
        payload.forEach { base[$0.key] = $0.value }
        return base
    }

    private func error(code: String, message: String) -> [String: Any] {
        return [
            "ok": false,
            "error": [
                "code": code,
                "message": message
            ]
        ]
    }
}
